src/service/viewport-impl.js

```diff
@@ -1093,22 +1137,37 @@ export class ViewportBindingNaturalIosEmbed_ {
   }
 
   /** @override */
-  updatePaddingTop(paddingTop, opt_updateScrollPos, opt_lastPaddingTop) {
+  hideViewerHeader(transient, lastPaddingTop) {
+    if (!transient) {
+      this.updatePaddingTop(0);
+    } else {
+      // Add extra paddingTop to make the content stay at the same position
+      // when the hiding header operation is transient
+      onDocumentReady(this.win.document, doc => {
+        const existingPaddingTop =
+            this.win./*OK*/getComputedStyle(doc.body)['padding-top'] || '0';
```

Comment:
Yeah, you are right. I didn't found this problem because when I'm testing, I am always calling `hideViewerHeader(true, '50px')` then `showViewerHeader(false, '50px')` then `hideViewerHeader(true, '50px')` ... where I will always reset the `body.style.paddingTop` inside `showViewerHeader(false, '50px')`. So the current code only works when we receive message like `hide(transient, 50px)`->`show(non-transient, 50px)`->`hide(transient, 50px)`->`show(non-transient, 50px)`-... It doesn't work if we receive two consecutive hide header messages that want to set the paddingTop differently (first hide message wants to set paddingTop to 20px temporarily, and second hide message wants to set paddingTop to 0px temporarily)

Let me think about it more on how to fix this.